### PR TITLE
Bug 1986225: add tests for vm snapshot feature

### PR DESCRIPTION
**Pull request by gouyang**

### CODE REVIEW

File: frontend/packages/kubevirt-plugin/integration-tests-cypress/tests/tabs/snapshot.ts [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+import * as snapshotView from '../../view/snapshot';
+import { tab } from '../../view/tab';
+import { virtualization } from '../../view/virtualization';
+import { vm } from '../../view/vm';
+
+const vmData: VirtualMachineData = {
+  name: `test-vm-snapshot-${testName}`,
+  description: 'rhel8 vm for snapshot',
+  namespace: testName,
+  template: 'Red Hat Enterprise Linux 8.0+ VM',
```
Comment (pcbailey) [inline frontend/packages/kubevirt-plugin/integration-tests-cypress/tests/tabs/snapshot.ts:15]: Do we have strings anywhere for the templates? If not, we should probably add them.

File: frontend/packages/kubevirt-plugin/integration-tests-cypress/tests/tabs/snapshot.ts [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+
+    // create snapshot
+    tab.navigateToSnapshot();
+    cy.get(snapshotView.takeSnapshot).click();
+    cy.get(snapshotView.snapshotName)
+      .clear()
+      .type(snapshotName);
+    cy.get(snapshotView.approve).click();
+    cy.get(snapshotView.save).click();
+    cy.get(snapshotView.status(snapshotName)).contains('Ready');
```
Comment (pcbailey) [inline frontend/packages/kubevirt-plugin/integration-tests-cypress/tests/tabs/snapshot.ts:63]: nitpick: 'Ready' should be a constant

File: frontend/packages/kubevirt-plugin/integration-tests-cypress/view/snapshot.ts [hunk 3]
```diff
@@ -0,0 +1,11 @@
+export const takeSnapshot = 'button[id="add-snapshot"]';
```
Comment (pcbailey) [inline frontend/packages/kubevirt-plugin/integration-tests-cypress/view/snapshot.ts:1]: Would you mind separating the functions from the constants and maybe add an empty line to make it easier to see that they're different? Could you also add `Button` or `Btn` to the end of the button selectors here that are missing it?

File: frontend/packages/kubevirt-plugin/integration-tests-cypress/view/selector.ts [hunk 4]
```diff
@@ -63,3 +63,6 @@ export enum storageClass {
   dropdown = 'button[id="form-ds-sc-select"]',
   selectMenu = '.pf-c-select__menu',
 }
+
+// yaml page
+export const yamlCreate = 'button[data-test="save-changes"]';
```
Comment (pcbailey) [inline frontend/packages/kubevirt-plugin/integration-tests-cypress/view/selector.ts:68]: I think we should be consistent and include `Button` or `Btn` at the end of all button selectors. I don't care which one we use. I just think it needs to be consistent. I think the verb should go first in this one, like `createYamlButton` or `createYamlBtn`.

File: frontend/packages/kubevirt-plugin/integration-tests-cypress/tests/tabs/snapshot.ts [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+      .click();
+
+    // create snapshot
+    tab.navigateToSnapshot();
+    cy.get(snapshotView.takeSnapshot).click();
+    cy.get(snapshotView.snapshotName)
+      .clear()
+      .type(snapshotName);
+    cy.get(snapshotView.approve).click();
+    cy.get(snapshotView.save).click();
```
Comment (pcbailey) [inline frontend/packages/kubevirt-plugin/integration-tests-cypress/tests/tabs/snapshot.ts:62]: Can we make this a function in the snapshots view?

File: frontend/packages/kubevirt-plugin/integration-tests-cypress/tests/tabs/snapshot.ts [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+    tab.navigateToDetails();
+    vm.stop(vmData);
+    tab.navigateToSnapshot();
+    cy.get('.pf-c-table__text')
+      .contains('Name')
+      .should('be.visible');
+
+    // restore snapshot
+    cy.get(snapshotView.restoreButton(snapshotName)).click();
+    cy.get(modalConfirm).click();
```
Comment (pcbailey) [inline frontend/packages/kubevirt-plugin/integration-tests-cypress/tests/tabs/snapshot.ts:77]: Can we make this a function in the snapshots view?

File: frontend/packages/kubevirt-plugin/integration-tests-cypress/tests/tabs/snapshot.ts [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+    cy.get(snapshotView.lastRestore(snapshotName)).contains('Just now');
+
+    // delete snapshot
+    cy.get(snapshotView.kebabButton).click();
+    cy.get(snapshotView.deleteButton).click();
+    cy.get(modalConfirm).click();
```
Comment (pcbailey) [inline frontend/packages/kubevirt-plugin/integration-tests-cypress/tests/tabs/snapshot.ts:83]: Can we make this a function in the snapshots view?

**Comment by gouyang** [comment]
@pcbailey I addressed all of your comments, could you review again?

**Comment by pcbailey** [comment]
/lgtm
This looks great!

**Pull request merged**